postgresql/endpoint_methods.go

```diff
@@ -0,0 +1,83 @@
+package postgresql
+
+import (
+	"github.com/lib/pq"
+
+	"github.com/benchttp/server/benchttp"
+)
+
+func (c ComputedStatsService) ListMetadataByUserID(userID int) ([]benchttp.Metadata, error) {
+	metadataList := []benchttp.Metadata{}
+
+	stmt, err := c.db.Prepare(`SELECT tag, finished_at FROM metadata WHERE user_id = $1 ORDER BY finished_at DESC`)
+	if err != nil {
+		return []benchttp.Metadata{}, ErrPreparingStmt
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query(userID)
+	if err != nil {
+		return []benchttp.Metadata{}, ErrExecutingPreparedStmt
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		metadata := benchttp.Metadata{}
+		err = rows.Scan(
+			&metadata.Tag,
+			&metadata.FinishedAt,
+		)
+		if err != nil {
+			return nil, err
+		}
+		metadataList = append(metadataList, metadata)
+	}
+
+	return metadataList, nil
+}
+
+func (c ComputedStatsService) FindComputedStatsByID(metadataID int) (benchttp.ComputedStats, error) {
+	computedStats := benchttp.ComputedStats{}
+
+	stmt := `SELECT
+			m.tag,
+			m.finished_at,
+			c.code_1xx,
+			c.code_2xx,
+			c.code_3xx,
+			c.code_4xx,
+			c.code_5xx,
+			t.min,
+			t.max,
+			t.mean,
+			t.median,
+			t.variance,
+			t.deciles
+		FROM public.metadata AS m
+		INNER JOIN public.codestats AS c ON c.metadata_id = m.id
+		INNER JOIN public.timestats AS t ON t.metadata_id = m.id
+		WHERE m.id = $1
+		ORDER BY m.finished_at DESC`
```

Comment:
[nitpick] If this works it's not that much of a problem, but this results in a weirdly indented query, because <code>``</code> keeps the formating:
```sql
SELECT
			m.tag,
			m.finished_at,
			-- ...
		FROM public.metadata AS m
		INNER JOIN public.codestats AS c ON c.metadata_id = m.id
		-- ...
```

That's why I suggested the following:
```go
	stmt := `
SELECT
	m.tag,
	m.finished_at,
	-- ...
FROM public.metadata AS m
INNER JOIN public.codestats AS c ON c.metadata_id = m.id
-- ...`[1:]
```
(Although missing indentation inside a function can seem confusing at first, it's actually very common to do so)

Reply:
We tried IRL with Greg, but with the deciles and our nested structs, it is difficult :(